tests/jax/components/building/networks_test.py

```diff
@@ -0,0 +1,200 @@
+        test_builder: Pytest fixture for test system builder
+
+    Returns:
+        None.
+    """
+    assert not hasattr(test_builder.store, "key")
+    test_default_networks.on_building_init_start(test_builder)
+    assert hasattr(test_builder.store, "key")
+    assert isinstance(test_builder.store.key, jax.random.PRNGKeyArray) or isinstance(
+        test_builder.store.key, jax.numpy.DeviceArray
+    )
+
+
+def test_config_set(test_default_networks: Networks) -> None:
+    """Test if config is set by component init.
+
+    Args:
+        test_default_networks: Pytest fixture for default networks component.
+
+    Returns:
+        None.
+    """
+    assert test_default_networks.config.seed == 919
+
+
+def test_network_factory_environment_spec(
+    test_default_networks: Networks, test_builder: SystemBuilder
+) -> None:
+    """Test if environment spec is given to the network factory and stored by build.
+
+    Args:
+        test_default_networks: Pytest fixture for default networks component.
+        test_builder: Pytest fixture for test system builder
+
+    Returns:
+        None.
+    """
+    test_default_networks.on_building_init_start(test_builder)
+    networks = test_builder.store.network_factory()
+
+    for network in networks.values():
+        assert network["environment_spec"]._keys == ["agent_0", "agent_1"]
+        assert network["environment_spec"]._specs["agent_0"].observations.shape == (
+            10,
+            5,
+        )
+
+
+def test_network_factory_agent_net_keys(
+    test_default_networks: Networks, test_builder: SystemBuilder
+) -> None:
+    """Test if net keys are given to the network factory and stored by build.
+
+    Args:
+        test_default_networks: Pytest fixture for default networks component.
+        test_builder: Pytest fixture for test system builder
+
+    Returns:
+        None.
+    """
+    test_default_networks.on_building_init_start(test_builder)
+    networks = test_builder.store.network_factory()
+
+    for network in networks.values():
+        assert network["agent_net_keys"] == {"key1": "value1", "key2": "value2"}
+
+
+def test_network_factory_rng_keys(
+    test_default_networks: Networks, test_builder: SystemBuilder
+) -> None:
+    """Test if rng keys are given to the network factory and stored by build.
+
+    Args:
+        test_default_networks: Pytest fixture for default networks component.
+        test_builder: Pytest fixture for test system builder
+
+    Returns:
+        None.
+    """
+    test_default_networks.on_building_init_start(test_builder)
+    networks = test_builder.store.network_factory()
+
+    keys = []
+    for network in networks.values():
+        assert isinstance(network["rng_key"], jax.random.PRNGKeyArray) or isinstance(
+            network["rng_key"], jax.numpy.DeviceArray
+        )
+        keys.append(tuple(network["rng_key"].tolist()))
+    assert len(set(keys)) == 1
```

Comment:
Not sure about the key handling, but does this mean that all networks should always share the same `rng_key`?

Reply:
Yes it does. I have added a comment to make this clearer.

Reply:
See #592. Will address in a separate PR